tasks/update_go.py

```diff
@@ -120,15 +126,106 @@ def _handle_file(path: str, patterns: Dict[re.Pattern, str], expected_match: int
+        _update_images(gover)
+        if local_uncommited_changes_exist(repo):
+            create_branch_and_push_changes(
+                repo,
+                branch,
+                f"Update Go version to {gover} for datadog agent buildimages",
+            )
+        else:
+            raise exceptions.Exit(f"Updating buildimages to {gover} changed nothing. Exiting.")
+
+        # gitlab job takes a bit to start up
+        # GIT_COMMIT_SHORT_SHA used in gitlab is first 8 characters of commit SHA
+        commit_sha = subprocess.check_output(["git", "rev-parse", "@"]).strip().decode('utf-8')
+        print(commit_sha)
+        tries = 10
+        build_ids = []
+        for i in range(tries):
+            status = requests.get(f"https://api.github.com/repos/DataDog/datadog-agent-buildimages/commits/{commit_sha}/status", headers={"Cache-Control": "no-cache"}).text
+            print(status)
+            matches = re.findall("\"https://gitlab.ddbuild.io/datadog/datadog-agent-buildimages/builds/(\d+)\"", status)
+            if len(matches) < 1:
+                if i < tries - 1:
+                    time.sleep(15)
+                    continue
+                else:
+                    raise exceptions.Exit("Exhausted retries waiting for gitlab job to start.")
+            else:
+                build_ids.extend(matches)
+                print(f"Gitlab job started with build_ids: {build_ids}")
+                break
+
+        # XXX: is it OK to require the user create a gitlab access token?
+        res = requests.get(f"https://gitlab.ddbuild.io/api/v4/projects/291/jobs/{build_ids[0]}", headers={"PRIVATE-TOKEN": os.environ["GITLAB_ACCESS_TOKEN"]})
+        pipeline_id = res.json()["pipeline"]["id"]
+        image_tag = f"v{pipeline_id}-{commit_sha[:8]}_test_only"
+        _create_agent_pr(branch, gover, image_tag)
+
+    create_pr_link = f"https://github.com/DataDog/datadog-agent-buildimages/compare/{branch}?expand=1"
+    print(f"Opening link to create an associated PR for the updated buildimages: {create_pr_link}")
+    subprocess.run(["open", create_pr_link])
+
+    create_pr_link = f"https://github.com/DataDog/datadog-agent/compare/{branch}?expand=1"
+    print(f"Opening link to create an associated PR for the agent with updated images: {create_pr_link}")
+    subprocess.run(["open", create_pr_link])
+
+
+def _create_agent_pr(branch, gover, image_tag):
+    repo = "datadog-agent"
+    with dd_repo_temp_cwd(repo):
+        if local_uncommited_changes_exist(repo):
+            raise "Uncommited changes exist in datadog-agent repo. Exiting."
+        checkout_latest_main(repo)
+        out = subprocess.check_output('inv', '-e', 'update-go', gover, image_tag)
+        if local_uncommited_changes_exist(repo):
+            create_branch_and_push_changes(
+                repo,
+                branch,
+                f"Update Go version to {gover} and buildimages to {image_tag} for datadog agent",
+            )
```

Comment:
Having a task which cd to a different repository to run some other commands (in particular creating a PR) seems too risky to me